drivers/led_strip/ws2812_uart.c

```diff
@@ -0,0 +1,333 @@
+						3 * frame_len),                                    \
+			WS2812_PACKED_UART_BYTE(((one_frame) << 2 * frame_len) |                   \
+							((one_frame) << frame_len) | (zero_frame), \
+						3 * frame_len),                                    \
+			WS2812_PACKED_UART_BYTE(((one_frame) << 2 * frame_len) |                   \
+							((one_frame) << frame_len) | (one_frame),  \
+						3 * frame_len),                                    \
+	}
+
+/*
+ * Macro to create a lookup table with repeated frames for WS2812 LEDs.
+ * This generates the appropriate lookup table based on the repeat count.
+ */
+#define WS2812_LOOK_UP_TABLE_LOOP(repeat_count, zero_frame, one_frame, frame_len)                  \
+	WS2812_##repeat_count##X_LOOK_UP_TABLE_PREPARE((zero_frame) & ((1 << frame_len) - 1),      \
+						       (one_frame) & ((1 << frame_len) - 1),       \
+						       frame_len)
+
+/*
+ * Helper macro to handle concatenation for lookup table creation.
+ */
+#define WS2812_LOOK_UP_TABLE_LOOP_PASTER(repeat_count, zero_frame, one_frame, frame_len)           \
+	WS2812_LOOK_UP_TABLE_LOOP(repeat_count, zero_frame, one_frame, frame_len)
+
+/*
+ * Macro to prepare the lookup table for WS2812 LEDs.
+ * Generates the lookup table based on the repeat count, zero frame, one frame, and frame length.
+ */
+#define WS2812_LOOK_UP_TABLE_PREPARE(repeat_count, zero_frame, one_frame, frame_len)               \
+	WS2812_LOOK_UP_TABLE_LOOP_PASTER(repeat_count, zero_frame, one_frame, frame_len)
+
+/* Lookup table formation macro definitions ends here*/
+
+#define WS2812_DEFAULT_RESET_DELAY_MS (50)
+#define WS2812_UART_NUM_PIXELS(idx)   (DT_INST_PROP(idx, chain_length))
+#define WS2812_UART_BUFSZ(idx, repeat)                                                             \
+	((WS2812_NUM_COLORS(idx) * 8 * WS2812_UART_NUM_PIXELS(idx)) / repeat)
+
+/*
+ * Retrieve the channel to color mapping (e.g. RGB, BGR, GRB, ...) from the
+ * "color-mapping" DT property.
+ */
+#define WS2812_COLOR_MAPPING(idx)                                                                  \
+	static const uint8_t ws2812_uart_##idx##_color_mapping[] = DT_INST_PROP(idx, color_mapping)
+
+#define WS2812_NUM_COLORS(idx) (DT_INST_PROP_LEN(idx, color_mapping))
+
+/* Get the latch/reset delay from the "reset-delay" DT property. */
+#define WS2812_RESET_DELAY(idx) DT_INST_PROP_OR(idx, reset_delay, WS2812_DEFAULT_RESET_DELAY_MS)
+
+#define UART_NODE(idx) DT_PARENT(DT_INST(idx, worldsemi_ws2812_uart))
+
+#define WS2812_UART_DEVICE(idx)                                                                    \
+                                                                                                   \
+	BUILD_ASSERT(DT_PROP(UART_NODE(idx), tx_invert),                                           \
+		     "This driver depends on tx-invert UART property and "                         \
+		     "is not set. Please check your device-tree settings");                        \
```

Comment:
I don't think this should assert, give a warning perhaps if it is not set, but it would be valid operation to use a normal UART that does not support this mode with an external single input invert logic gate

Reply:
Looked around and unable to find a way to print warning based on condition in macro expansion.
Can you please provide some reference and guidance

Reply:
It would need to be done through cmake e.g. `message(WARNING "message here")`

Reply:
![image](https://github.com/user-attachments/assets/1c101223-963b-4575-a5c7-6fdf55d2b17f)